src/VenueModel.cpp

```diff
@@ -282,15 +244,28 @@ QVariant VenueModel::data(const QModelIndex &index, int role) const
 
         auto const openingMinutes = openingMinutesVar.toList();
 
-        switch(role) {
+        switch(role)
+        {
         case VenueModelRoles::Open:
             return isInRange(openingMinutes[m_currendDayIndex].toMap(), m_currentMinute);
         case VenueModelRoles::ClosesSoon:
             // ClosesSoon holds true if venue is NOT open in half an hour from now
             return !isInRange(openingMinutes[m_currendDayIndex].toMap(), m_currentMinute + MINUTES_CLOSES_SOON);
         }
     }
+    break;
+    case VenueModelRoles::CondensedOpeningHours:
+    {
+        const auto openingHoursVar = QStandardItemModel::data(index, VenueModel::OpeningHours);
+        if (!openingHoursVar.isValid())
+        {
+            return QVariant::Invalid;
+        }
 
-    return QStandardItemModel::data(index, role);
+        return condenseOpeningHours(openingHoursVar.toList(), m_currendDayIndex);
+    }
+    default:
+        return QStandardItemModel::data(index, role);
+    }
 }
```

Comment:
VenueModel.cpp:270: warning: control may reach end of non-void function [-Wreturn-type]